src/lambda/vue/samInvoke.ts

```diff
@@ -0,0 +1,501 @@
+        items: pickerItems,
+        options: {
+            title: localize('AWS.lambda.form.selectDebugConfig', 'Select Debug Configuration'),
+        },
+    })
+
+    const choices = await picker.promptUser({
+        picker: qp,
+    })
+    const pickerResponse = picker.verifySinglePickerOutput<LaunchConfigPickItem>(choices)
+
+    if (!pickerResponse) {
+        return
+    }
+
+    if (pickerResponse.index === -1) {
+        const ib = input.createInputBox({
+            options: {
+                prompt: localize('AWS.lambda.form.debugConfigName', 'Enter Name For Debug Configuration'),
+            },
+        })
+        const response = await input.promptUser({ inputBox: ib })
+        if (response) {
+            launchConfig.addDebugConfiguration(finalizeConfig(config, response))
+            await openLaunchJsonFile()
+        }
+    } else {
+        // use existing label
+        launchConfig.editDebugConfiguration(finalizeConfig(config, pickerResponse.label), pickerResponse.index)
+        await openLaunchJsonFile()
+    }
+}
+
+/**
+ * Validate and execute the provided launch config.
+ * TODO: Post validation failures back to webview?
+ * @param config Config to invoke
+ */
+async function invokeLaunchConfig(config: AwsSamDebuggerConfiguration, context: ExtContext): Promise<void> {
+    const finalConfig = finalizeConfig(config, 'Editor-Created Debug Config')
+
+    const targetUri = getUriFromLaunchConfig(finalConfig)
+
+    const folder = targetUri ? vscode.workspace.getWorkspaceFolder(targetUri) : undefined
+
+    await new SamDebugConfigProvider(context).resolveDebugConfiguration(folder, finalConfig)
+}
+
+function getUriFromLaunchConfig(config: AwsSamDebuggerConfiguration): vscode.Uri | undefined {
+    let targetPath: string
+    if (isTemplateTargetProperties(config.invokeTarget)) {
+        targetPath = config.invokeTarget.templatePath
+    } else if (isCodeTargetProperties(config.invokeTarget)) {
+        targetPath = config.invokeTarget.projectRoot
+    } else {
+        // error
+        return undefined
+    }
+    if (path.isAbsolute(targetPath)) {
+        return vscode.Uri.file(targetPath)
+    }
+    const workspaceFolders = vscode.workspace.workspaceFolders || []
+    for (const workspaceFolder of workspaceFolders) {
+        const absolutePath = tryGetAbsolutePath(workspaceFolder, targetPath)
+        if (fs.pathExistsSync(absolutePath)) {
+            return vscode.Uri.file(absolutePath)
+        }
+    }
+
+    return undefined
+}
+
+function makeSampleRequestManifestResourceFetcher(): ResourceFetcher {
+    return new CompositeResourceFetcher(
+        new HttpResourceFetcher(sampleRequestManifestPath, { showUrl: true }),
+        new FileResourceFetcher(ext.manifestPaths.lambdaSampleRequests)
+    )
+}
+
+function getLaunchConfigQuickPickItems(launchConfig: LaunchConfiguration, uri: vscode.Uri): LaunchConfigPickItem[] {
+    const existingConfigs = launchConfig.getDebugConfigurations()
```

Comment:
`getDebugConfigurations()` could be modified to take a `valid: boolean` parameter, so that it does what's wanted here.